File: pixsim7/backend/main/api/v1/dev_app_map.py
"""
App Map Snapshot API (v2).

Provides the canonical backend-served App Map snapshot for dev tooling.
"""

from typing import Optional

from fastapi import APIRouter, Depends

from pixsim7.backend.main.api.dependencies import get_current_user_optional
from pixsim7.backend.main.domain.user import User

from .dev_app_map_contract import AppMapSnapshotV2
from .dev_app_map_service import build_app_map_snapshot


router = APIRouter(prefix="/dev/app-map", tags=["dev"])


@router.get("/snapshot", response_model=AppMapSnapshotV2)
async def get_app_map_snapshot(
    user: Optional[User] = Depends(get_current_user_optional),
):
    """
    Return canonical App Map snapshot (v2).

    Includes frontend generated registry data, backend runtime introspection,
    and optional external registry references.
    """
    return build_app_map_snapshot()

File: packages/shared/api/client/src/domains/devAppMap.ts
/**
 * Dev App Map API Domain Client
 *
 * Client for App Map snapshot v2 endpoints.
 */

import type { AppMapSnapshotV2 } from '@pixsim7/shared.types';
import type { PixSimApiClient } from '../client';

export type { AppMapSnapshotV2 };

export function createDevAppMapApi(client: PixSimApiClient) {
  return {
    async getSnapshot(): Promise<AppMapSnapshotV2> {
      return client.get<AppMapSnapshotV2>('/dev/app-map/snapshot');
    },
  };
}
